api/src/main/java/ca/bc/gov/educ/api/trax/controller/v2/SchoolController.java

```diff
@@ -122,12 +122,27 @@ public ResponseEntity<List<SchoolDetail>> getSchoolsBySchoolCategory(@RequestPar
         return response.GET(schoolService.getSchoolDetailsBySchoolCategoryCode(schoolCategoryCode));
     }
 
-    @GetMapping(EducGradTraxApiConstants.GRAD_SCHOOL_DETAIL_URL_MAPPING_V2 + EducGradTraxApiConstants.GET_SCHOOL_BY_CODE_MAPPING)
+    @GetMapping(EducGradTraxApiConstants.GRAD_SCHOOL_DETAIL_URL_MAPPING_V2 + EducGradTraxApiConstants.GET_SCHOOL_BY_SCHOOL_ID)
+    @PreAuthorize(PermissionsConstants.READ_SCHOOL_DATA)
+    @Operation(summary = "Find School Details by ID from cache", description = "Get School Details by ID from cache", tags = { "School" })
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "204", description = "NO CONTENT")})
+    public ResponseEntity<SchoolDetail> getSchoolDetailsById(@PathVariable UUID schoolId) {
+        log.debug("getSchoolDetails V2 : ");
+        SchoolDetail schoolDetailResponse = schoolService.getSchoolDetailBySchoolId(schoolId);
+        if(schoolDetailResponse != null) {
+            return response.GET(schoolDetailResponse);
+        }else {
+            return response.NOT_FOUND();
+        }
+    }
+
+    @GetMapping(EducGradTraxApiConstants.GRAD_SCHOOL_DETAIL_URL_MAPPING_V2 + EducGradTraxApiConstants.GET_SCHOOL_DETAIL_SEARCH_MAPPING)
     @PreAuthorize(PermissionsConstants.READ_SCHOOL_DATA)
     @Operation(summary = "Find School Details by Mincode from cache", description = "Get School Details by Mincode from cache", tags = { "School" })
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "OK"),
             @ApiResponse(responseCode = "204", description = "NO CONTENT")})
-    public ResponseEntity<SchoolDetail> getSchoolDetailsByMincode(@PathVariable String minCode) {
+    public ResponseEntity<SchoolDetail> getSchoolDetailsByParams(@RequestParam(required = false) String minCode) {
         log.debug("getSchoolDetails V2 : ");
```

Comment:
Just minor... the same debug as school by schoolId at line 131 makes me hard to trace...